pkgs/applications/networking/cluster/terraform-providers/default.nix

```diff
@@ -22,133 +39,20 @@ let
       passthru = data;
     };
 
-  # Google is now using the vendored go modules, which works a bit differently
-  # and is not 100% compatible with the pre-modules vendored folders.
-  #
-  # Instead of switching to goModules which requires a goModSha256, patch the
-  # goPackage derivation so it can install the top-level.
-  patchGoModVendor = drv:
-    drv.overrideAttrs (attrs: {
-      buildFlags = "-mod=vendor";
-
-      # override configurePhase to not move the source into GOPATH
-      configurePhase = ''
-        export GOPATH=$NIX_BUILD_TOP/go:$GOPATH
-        export GOCACHE=$TMPDIR/go-cache
-        export GO111MODULE=on
-      '';
-
-      # just build and install into $GOPATH/bin
-      buildPhase = ''
-        go install -mod=vendor -v -p 16 .
-
-        runHook postBuild
-      '';
-
-      # don't run the tests, they are broken in this setup
-      doCheck = false;
-    });
-
   # These providers are managed with the ./update-all script
-  automated-providers = lib.mapAttrs (toDrv) list;
+  automated-providers = lib.mapAttrs (name: attrs:
+    (if attrs ? vendorSha256 then buildWithGoModule else buildWithGoPackage)
+      name attrs) list;
```

Comment:
I don't think this is right. shouldn't it be:
```suggestion
    (if (lib.hasAttr "vendorSha256" attrs) then buildWithGoModule else buildWithGoPackage)
      name attrs) list;
```

Reply:
This is some legit nix ([source](https://nixos.org/manual/nix/stable/#sec-language-operators)) :wink: 

![image](https://user-images.githubusercontent.com/1334474/99703219-bf693d80-2a8e-11eb-8929-7873ec510411.png)

More than happy to use the `lib.hasAttr` function instead of the `?` operator if that is the preferred way.

Reply:
oh, well it seemed to have the expected behavior, but I think `lib.hasAttr` is more readable. and isn't contextual like `?` which can also be used to provide a default value when inside an attrset.

Reply:
Indeed, the function is the least ambiguous of the two. Fixed it now.